File: backend/Foody/src/main/java/com/foody/recipe/entity/Recipe.java
package com.foody.recipe.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor // openCsv때문에 열어놔야함
@AllArgsConstructor
@Getter
@Entity(name = "recipe")
public class Recipe {

    @Id private long id;
    @Column private String name;
    @Column(columnDefinition = "TEXT") private String ingredient;
    @Column(columnDefinition = "TEXT") private String description;
    @Column private String url;
    @Column private String difficulty;
    @Column private double servers;
    @Column private String foodMethod;
    @Column private String foodSituation;
    @Column private String foodIngredients;
    @Column private String foodTypes;
    @Column private double energy;
    @Column private double carbohydrates;
    @Column private double protein;
    @Column private double dietaryFiber;
    @Column private double calcium;
    @Column private double sodium;
    @Column private double iron;
    @Column private double fats;
    @Column private double vitaminA;
    @Column private double vitaminC;
}

File: bigdata/Foody/src/main/kotlin/com/data/foody/domain/Recipe.kt
package com.data.foody.domain

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true) //  JSON 직렬화/역직렬화 중에 알려지지 않은 속성이 있을 경우 무시
data class Recipe(
        val recipe_id: Long,
        val ingredients: String,
        val ingredients_concat: String,
        val ingredient_count: Int,
        val energy: Float,
        val carbohydrates: Float,
        val protein: Float,
        val fats: Float,
        val dietaryFiber: Float,
        val calcium: Float,
        val sodium: Float,
        val iron: Float,
        val vitaminA: Float,
        val vitaminC: Float,
        val difficulty: String,
        val servers: Int,
        val food_preparation_methods: String,
        val food_situations: String,
        val food_ingredients: String,
        val food_types: String
)
